test/dtypes/test_nf4.py

```diff
@@ -165,44 +179,50 @@ def test_load_from_nf4_diff_meta(self):
         assert other_mod.param.block_size == 64
         assert other_mod.param.scaler_block_size == 1
 
-    def test_to_copy(self):
+    @parametrize("dtype", [torch.bfloat16, torch.float16, torch.float32])
+    def test_to_copy(self, dtype: torch.dtype):
         inpt_tensor = torch.rand(128, device='cpu')
         inpt_tensor_nf4 = to_nf4(inpt_tensor, 32, 2)
-        inpt_tensor_bfloat16 = inpt_tensor_nf4.to(torch.bfloat16)
-        torch.testing.assert_allclose(inpt_tensor, inpt_tensor_bfloat16, atol=0.13, rtol=0.13)
+        nf4_to_dtype = inpt_tensor_nf4.to(dtype)
+        torch.testing.assert_allclose(inpt_tensor, nf4_to_dtype, atol=0.13, rtol=0.13)
 
         if torch.cuda.is_available():
             inpt_tensor = torch.rand(128, device='cuda')
             inpt_tensor_nf4 = to_nf4(inpt_tensor, 32, 2)
-            inpt_tensor_bfloat16 = inpt_tensor_nf4.to(torch.bfloat16)
-            torch.testing.assert_allclose(inpt_tensor, inpt_tensor_bfloat16, atol=0.13, rtol=0.13)
+            nf4_to_dtype = inpt_tensor_nf4.to(dtype)
+            torch.testing.assert_allclose(inpt_tensor, nf4_to_dtype, atol=0.13, rtol=0.13)
 
-    def test_to_bfloat16(self):
-        inpt_tensor = torch.rand(128, dtype=torch.bfloat16)
+    @parametrize("dtype", [torch.bfloat16, torch.float16, torch.float32])
+    def test_to_dtype(self, dtype: torch.dtype):
+        inpt_tensor = torch.rand(128, dtype=dtype)
         inpt_tensor_nf4 = to_nf4(inpt_tensor, 32, 2)
         assert type(inpt_tensor_nf4) != torch.Tensor
-        assert type(inpt_tensor_nf4.to(torch.bfloat16)) == torch.Tensor
-        assert inpt_tensor_nf4.to(torch.bfloat16).dtype == torch.bfloat16
+        assert type(inpt_tensor_nf4.to(dtype)) == torch.Tensor
+        assert inpt_tensor_nf4.to(dtype).dtype == dtype
 
     @unittest.skipIf(not torch.cuda.is_available(), "Need CUDA available")
-    def test_smoketest_linear(self):
-        a = torch.randn(32, 32, dtype=torch.bfloat16, device='cuda')
+    @parametrize("dtype", [torch.bfloat16, torch.float16, torch.float32])
+    def test_smoketest_linear(self, dtype: torch.dtype):
+        a = torch.randn(32, 32, dtype=dtype, device='cuda')
         a_nf4 = torchao.dtypes.to_nf4(a, 16, 2)
         inp = torch.randn(2, 32, 32, dtype=a.dtype, device=a.device)
         out1 = torch.nn.functional.linear(inp, a)
         out2 = torch.nn.functional.linear(inp, a_nf4)
 
     @unittest.skipIf(not torch.cuda.is_available(), "Need CUDA available")
-    def test_smoketest_linear_compile(self):
-        for dtype in [torch.bfloat16, torch.float16]:
-            if torch.cuda.is_available() and torch.cuda.get_device_capability() < (8, 0) and dtype == torch.bfloat16:
-                self.skipTest("test requires SM capability of at least (8, 0).")
```

Comment:
this test is always skipped before because we exit with `self.skipTest` with `torch.bfloat16` and did not have a chance to test `torch.float16`